### PR TITLE
feat: Add active debuggee support.

### PR DESCRIPTION
**Pull request by jasonborg**
The Firebase client now adds two new fields to the Debuggee entry in
the Firebase RTDB, `registrationTimeMsec` and `lastUpdateTimeMsec`.
The first time the Debuggee entry is written to the RTDB, both fields
are set. Thereafter only the `lastUpdateTimeMsec` will be changed, which
is done periodically every hour when the agent is running continuously,
it is also updated when the agent first starts up

This is in support of
https://github.com/GoogleCloudPlatform/snapshot-debugger/issues/76.

### CODE REVIEW

File: src/agent/internals/src/main/java/com/google/devtools/cdbg/debuglets/java/FirebaseClient.java [hunk 1]
```diff
@@ -991,45 +1094,42 @@ private static Object getDbValue(
...
-            @Override
-            public void onDataChange(DataSnapshot dataSnapshot) {
-              Object value = dataSnapshot.getValue();
-              infofmt(
-                  "Response obtained, data was %s found at %s", value == null ? "not" : "", path);
-
-              obtainedValue.add(value);
-              resultObtained.offer(Boolean.TRUE);
-            }
+    dbRef.addListenerForSingleValueEvent(
```
Comment (mctavish) [inline src/agent/internals/src/main/java/com/google/devtools/cdbg/debuglets/java/FirebaseClient.java:1100]: Much nicer :)

File: src/agent/internals/src/main/java/com/google/devtools/cdbg/debuglets/java/FirebaseClient.java [hunk 2]
```diff
@@ -991,45 +1094,42 @@ private static Object getDbValue(
...
+    // null will be returned on read timeout.
+    if (isSuccess == null) {
+      infofmt("Read from %s timed out after %d %s", path, timeout.value, timeout.units);
+      throw new Exception(
+          "Firebase Database read operation from '"
+              + path
+              + "' timed out after "
+              + timeout.value
+              + " "
+              + timeout.units.toString());
```
Comment (mctavish) [inline src/agent/internals/src/main/java/com/google/devtools/cdbg/debuglets/java/FirebaseClient.java:1130]: Or just add a toString() to the Timeout class.

File: src/agent/internals/src/main/java/com/google/devtools/cdbg/debuglets/java/FirebaseClient.java [hunk 3]
```diff
@@ -991,45 +1094,42 @@ private static Object getDbValue(
...
+    } else if (!isSuccess) {
+      throw new Exception("Error occurred attempting to read from the DB");
```
Comment (mctavish) [inline src/agent/internals/src/main/java/com/google/devtools/cdbg/debuglets/java/FirebaseClient.java:1132]: Nit: generally prefer a more specific subclass of Exception.

**Review by mctavish — APPROVED** [review]
Looks good.

File: src/agent/internals/src/main/java/com/google/devtools/cdbg/debuglets/java/FirebaseClient.java (hunk 2 above)
Comment (jasonborg) [reply src/agent/internals/src/main/java/com/google/devtools/cdbg/debuglets/java/FirebaseClient.java:1130]: good call

**Pull request merged**